src/platform/errors/errorUtils.ts

```diff
@@ -423,29 +435,73 @@ export function analyzeKernelErrors(
             };
         }
     }
-    if (lastTwolinesOfError && lastTwolinesOfError[1].toLowerCase().startsWith('ModuleNotFoundError'.toLowerCase())) {
-        const info = extractModuleAndFileFromImportError(lastTwolinesOfError[1]);
-        if (info) {
+    // This happens when ipykernel is not installed and we attempt to run without checking for ipykernel.
+    // '/home/don/samples/pySamples/crap/.venv/bin/python: No module named ipykernel_launcher\n'
+    const noModule = 'No module named'.toLowerCase();
+    const isNotAPackage = `is not a package`.toLowerCase();
+    if (stdErr.includes(noModule) && !isNotAPackage) {
+        const line = stdErrOrStackTrace
+            .splitLines()
+            .map((line) => line.trim())
+            .filter((line) => line.length)
+            .find((line) => line.toLowerCase().includes(noModule));
+        const moduleName = line ? line.substring(line.toLowerCase().indexOf(noModule) + noModule.length).trim() : '';
+        if (line) {
             return {
                 reason: KernelFailureReason.moduleNotFoundFailure,
-                moduleName: info.moduleName,
-                message: DataScience.failedToStartKernelDueToMissingModule().format(info.moduleName),
+                moduleName,
+                message: DataScience.failedToStartKernelDueToMissingModule().format(moduleName),
                 moreInfoLink: 'https://aka.ms/kernelFailuresMissingModule',
                 telemetrySafeTags: ['module.notfound.error']
             };
         }
-    }
-    // This happens when ipykernel is not installed and we attempt to run without checking for ipykernel.
-    // '/home/don/samples/pySamples/crap/.venv/bin/python: No module named ipykernel_launcher\n'
-    const noModule = 'No module named'.toLowerCase();
-    if (stdErr.includes(noModule)) {
+    } else if (stdErr.includes(noModule) && isNotAPackage) {
         const line = stdErrOrStackTrace
             .splitLines()
             .map((line) => line.trim())
             .filter((line) => line.length)
             .find((line) => line.toLowerCase().includes(noModule));
-        const moduleName = line ? line.substring(line.toLowerCase().indexOf(noModule) + noModule.length).trim() : '';
-        if (line) {
+        let moduleName = line ? line.substring(line.toLowerCase().indexOf(noModule) + noModule.length).trim() : '';
+        let fileName = '';
+        let folderName = '';
+        if (moduleName.split("'").length > 2) {
+            fileName = moduleName.split("'").slice(-2)[0] || '';
+            fileName = fileName ? `${fileName}.py` : '';
+            folderName = fileName ? fileName : '';
+            moduleName = moduleName.split("'")[1] || moduleName;
+        }
+        const modulesInCwd = filesInCwd
+            .filter((file) => file.toString().toLowerCase().endsWith('__init__.py'))
```

Comment:
Kinda pedantic of me, but seems like this should be an exact match and not endsWith. I don't really see someone naming a file HELLO__init__.py but you never know :D

Reply:
Fixed